libc-bottom-half/headers/public/wasi/api.h

```diff
@@ -31,6 +31,9 @@ _Static_assert(_Alignof(int32_t) == 4, "non-wasi data layout");
 _Static_assert(_Alignof(uint32_t) == 4, "non-wasi data layout");
 _Static_assert(_Alignof(int64_t) == 8, "non-wasi data layout");
 _Static_assert(_Alignof(uint64_t) == 8, "non-wasi data layout");
+// This header is only valid for wasm32: witx currently calcualtes memory
+// layout for the wasm32 abi only.
```

Comment:
No need to block this PR on this but maybe this is better of at the very top of the header and can be even more explicit. Alongside the `ifndef __wasi__` you could add `ifndef __wasm32__.. this header is only meant for wasm32`

Reply:
Then no comment needed just a `#error`

Reply:
For maximal clarity, I suggest phrasing this as "doesn't yet support wasm64" so that it's clear that this is just the present status, not a fundamental limitation.

Reply:
Implemented these suggestions in af28fd5